internal/validation/validation.go

```diff
@@ -699,6 +731,11 @@ func validateValueType(c *opContext, v common.Literal, t common.Type) (bool, str
 				if _, ok := t2.(*common.NonNull); !ok && v2.Default != nil {
 					t2 = &common.NonNull{OfType: t2}
 				}
+				if err == nil {
```

Comment:
This is error-prone, I would write this as:
if err == nil {
  if binding, ok := varBinding(c, v2.Name.Name); ok {
    return validateValueType(c, binding, t2)
  }
  if !typeCanBeUsedAs(t2, t) {
     ....
  }
}

or:

if err != nil {
  continue
}

Reply:
I have left CanBeUsedAs-part intact to minimise the diff. Of your two options, the second one looks healthier.

Reply:
`if err != nil {
    continue
}`